### PR TITLE
Maximum padded length errors

### PR DESCRIPTION
**Pull request by Adamits**
Addresses #40 

Adds the length comparison and message as a callback, which the padded tensor calls.

I also snuck in a little fix to `datasets.Item`.

### CODE REVIEW

File: yoyodyne/datasets.py [hunk 1]
```diff
@@ -41,8 +41,8 @@ def has_features(self):
         return self.features is not None
 
     @property
-    def has_targets(self):
-        return self.targets is not None
+    def has_target(self):
```
Comment (kylebgorman) [inline yoyodyne/datasets.py:44]: good catch

**Review by kylebgorman — APPROVED** [review]
LGTM.

Honestly, this design is kind of fun. Nice use of callbacks.

Can you think of any arguments for it over the other one? Should we ask Travis?

**Comment by Adamits** [comment]
> Honestly, this design is kind of fun. Nice use of callbacks.
Haha I thought it was fun too. Thanks.

> Can you think of any arguments for it over the other one? Should we ask Travis?
Hmmm, I think this one avoids an extra loop. Let's ask @bonham79 what he thinks between this and #51

**Comment by bonham79** [comment]
@Adamits @kylebgorman Howdy y'all, (sorry, github didn't flag my name getting called). I approve of this version more than the former. It's cleaner and keeps track of possible issues in a less head scratcy way.

**Pull request merged**